pymobiledevice3/services/afc.py

```diff
@@ -208,36 +208,43 @@ def __init__(self, lockdown: LockdownClient, service_name='com.apple.afc'):
         super().__init__(lockdown, service_name)
         self.packet_num = 0
 
-    def pull(self, src: str, dst: str, callback=None):
+    def pull(self, src: str, dst: str, callback: Optional[callable] = None):
         """ pull a file/directory from given path into a local file/directory """
-        if callback is not None:
-            callback(src, dst)
-
         src = self.resolve_path(src)
-
         if not self.isdir(src):
             # normal file
-            if os.path.isdir(dst):
-                dst = os.path.join(dst, os.path.basename(src))
-            with open(dst, 'wb') as f:
-                f.write(self.get_file_contents(src))
+            if dst.endswith(posixpath.sep):
```

Comment:
`os.sep`?

Reply:
wow... i missed it, i think there is bug here on windows (also if i will put `os.path.sep`).

need to check it, thanks!